### PR TITLE
NF: comment and simplify DismissNote

### PR DESCRIPTION
**Pull request by Arthur-Milchior**

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java [hunk 1]
```diff
@@ -450,22 +450,43 @@ public UndoDeleteNote(Note note, ArrayList<Card> allCs, @NonNull Card card) {
         }
     }
 
+
+
+    /**
+     * Represents an action that remove a card from the Reviewer without reviewing it.
+     *
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java:457]:
```suggestion
     * Represents an action that removes a card from the Reviewer without reviewing it.
```

**Review by david-allison-1 — COMMENTED** [review]
Flagging this for a later review - assumed it would be a simple change to review

File: AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java [hunk 2]
```diff
@@ -512,15 +533,15 @@ public SuspendCard(Card card) {
         }
 
         @Override
-        protected void actualTask(Collection col, Card card) {
+        protected void actualTask(Collection col) {
             // collect undo information
-            Card suspendedCard = card.clone();
+            MCard suspendedCard = mCard.clone();
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java:538]:
```suggestion
            Card suspendedCard = mCard.clone();
```

File: AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java [hunk 3]
```diff
@@ -512,15 +533,15 @@ public SuspendCard(Card card) {
...
             col.markUndo(new UndoSuspendCard(suspendedCard));
-            // suspend card
-            if (card.getQueue() == Consts.QUEUE_TYPE_SUSPENDED) {
-                col.getSched().unsuspendCards(new long[] {card.getId()});
+            // suspend mCard
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java:540]:
```suggestion
            // suspend card
```

File: AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java [hunk 4]
```diff
@@ -450,22 +450,43 @@ public UndoDeleteNote(Note note, ArrayList<Card> allCs, @NonNull Card card) {
...
+         * @param card The card that was in the reviewer. It usually is cloned and then restored if the action is undone
+         */
         public DismissNote(Card card) {
             this.mCard = card;
         }
 
-        protected abstract void actualTask(Collection col, Card card);
 
+        /**
+         * The art of the task that is specific to this object. E.g. suspending, deleting, burying...
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java:472]:
```suggestion
         * The task that is specific to this object. E.g. suspending, deleting, burying...
```

File: AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java [hunk 5]
```diff
@@ -450,22 +450,43 @@ public UndoDeleteNote(Note note, ArrayList<Card> allCs, @NonNull Card card) {
...
+         * @param col The collection
+         *
+         */
+        protected abstract void actualTask(Collection col);
 
+
+        /**
+         * @param col
+         * @param collectionTask A listener that potentially waits for the next card to review, sent as progress.
+         *                       Cancelling only means that the next card don't have to be sent, however the action should still be peformed.
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java:482]: I struggle to understand this. Could you consider revising

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
Doesn't compile

File: AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java (hunk 4 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java:472]: the Part of the task

It was just missing a letter. Thanks

File: AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java (hunk 5 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java]: Done

**Review by david-allison-1 — APPROVED** [review]
Thanks! LGTM

**Review by mikehardy — APPROVED** [review]
Works for me - thanks @Arthur-Milchior !

**Pull request merged**